hooks/read-package-hook/src/createPeerDependencyPatcher.ts

```diff
@@ -44,6 +55,19 @@ export function createPeerDependencyPatcher (
 
       pkg.peerDependencies![peerName] = currentVersions.join(' || ')
     }
+
+    const peerDepsOverrides = overrides.filter((override) => override.parentPkg && override.parentPkg.name === pkg.name)
+
+    peerDepsOverrides.forEach(override => {
+      if (!pkg.peerDependencies) pkg.peerDependencies = {}
+      const pkgVer = pkg.version ?? ''
+
+      if (!override.parentPkg!.pref ||
+        (isSubRange(override.parentPkg!.pref, pkgVer) || semver.satisfies(pkgVer, override.parentPkg!.pref))) {
+        pkg.peerDependencies[override.targetPkg.name] = override.newPref
+      }
+    })
```

Comment:
I don't understand. Why a peer dependency is now added even if it doesn't exist?

Reply:
In production there probably aren’t many use cases for it. I just felt that the option behaving the same in all cases made understanding it more trivial. But your point about not needing it if the package has no peer dependencies is very reasonable. I’ll fix that up